lib/parslet/atoms/can_flatten.rb

```diff
@@ -120,7 +120,7 @@ def flatten_repetition(list, named)
       return [] if named && list.empty?
 
       # If there are only strings, concatenate them and return that. 
-      foldl(list) { |s,e| s+e }
+      foldl(list) { |s,e| s.nil? ? e : e.nil? ? s : s + e }
```

Comment:
Can you special case the `?` by adding two if-statements above this line and leaving it alone? Using `?` cascaded hurts my sensibilities. ;)